app/assets/stylesheets/common/base/login-signup-page.scss

```diff
@@ -0,0 +1,304 @@
+// Shared styles
+
+.login-fullpage,
+.signup-fullpage,
+.invites-show {
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+
+  .signup-body,
+  .login-body {
+    display: flex;
+    justify-content: center;
+    width: 100%;
+    max-width: 800px;
+  }
+
+  .login-page-cta,
+  .signup-page-cta {
+    display: flex;
+    flex-direction: column;
+
+    &__existing-account,
+    &__no-account-yet {
+      color: var(--primary-medium);
+    }
+
+    &__disclaimer {
+      color: var(--primary-medium);
+      margin-bottom: 1rem;
+    }
+  }
+
+  .login-left-side {
+    box-sizing: border-box;
+    width: 100%;
+    padding: 3rem;
+    overflow: auto;
+    width: 100%;
+    max-width: 500px;
+  }
+
+  .login-right-side {
+    padding: 3rem;
+    position: relative;
+
+    &::before {
+      content: "";
+      position: absolute;
+      left: 0;
+      top: 50%;
+      width: 1px;
+      height: 80%;
+      transform: translateY(-50%);
+      background-color: var(--primary-low);
+    }
+  }
+
+  .input-group {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    margin-bottom: 0.8em;
+
+    input,
+    .select-kit-header {
+      padding: 0.75em 0.77em;
+      min-width: 250px;
+      margin-bottom: 0.25em;
+      width: 100%;
+    }
+
+    input:focus {
+      outline: none;
+      border: 1px solid var(--tertiary);
+      box-shadow: 0 0 0 2px rgba(var(--tertiary-rgb), 0.25);
+    }
+
+    input:disabled {
+      background-color: var(--primary-low);
+    }
+
+    span.more-info {
+      color: var(--primary-medium);
+      min-height: 1.4em; // prevents height increase due to tips
+      overflow-wrap: anywhere;
+    }
+
+    label.alt-placeholder,
+    .user-field.text label.control-label,
+    .user-field.dropdown label.control-label,
+    .user-field.multiselect label.control-label {
+      color: var(--primary-medium);
+      font-size: 16px;
+      font-weight: normal;
+      position: absolute;
+      pointer-events: none;
+      left: 1em;
+      top: 12px;
+      box-shadow: 0 0 0 0px rgba(var(--tertiary-rgb), 0);
+      transition: 0.2s ease all;
+    }
+
+    .user-field.text label.control-label,
+    .user-field.dropdown label.control-label,
+    .user-field.multiselect label.control-label {
+      z-index: 999;
+      top: -8px;
+      left: calc(1em - 0.25em);
+      background-color: var(--secondary);
+      padding: 0 0.25em 0 0.25em;
+      font-size: $font-down-1;
+    }
+
+    .user-field.text:focus-within,
+    .user-field.dropdown:focus-within,
+    .user-field.multiselect:focus-within {
+      z-index: 1000; // ensures the active input is always on top of sibling input labels
```

Comment:
We usually use the `z()` function, defined in common/foundation/variables.scss. Might not be necessary here, though (was this css moved/copied from somewhere else anyway?).

Reply:
This comes from: https://github.com/discourse/discourse/blob/db8e0a02f13aedffcbdc2077f919997787b9dd7f/app/assets/stylesheets/common/modal/login-modal.scss#L265-L269
which was introduced in https://github.com/discourse/discourse/pull/23967

I guess the z-index could be lower now that it's not a modal, but what layer would you use in this case?

https://github.com/discourse/discourse/blob/db8e0a02f13aedffcbdc2077f919997787b9dd7f/app/assets/stylesheets/common/foundation/variables.scss#L82-L109

Reply:
I'm not sure... I suspect actually that even `z-index: 1` will work here, the goal is just to show above sibling and that should do it... either way, fine to leave as is.